test/smtcallback.js

```diff
@@ -34,7 +34,12 @@ function expectErrors (errors, expectations) {
     if (errors[i].includes('Error trying to invoke SMT solver') || expectations[i].includes('Error trying to invoke SMT solver')) {
       continue;
     }
-    expectations[i] = expectations[i].replace('happens here.', 'happens here');
+    expectations[i] = expectations[i].replace('happens here.', 'happens here').split('\\n').join('\n');
+    errors[i] = errors[i].replace('happens here.', 'happens here').split('\\n').join('\n');
+    if (ignoreCex) {
+      expectations[i] = expectations[i].split('\nCounterexample')[0];
```

Comment:
If we do this, do we actually need to do that split outside of this branch?

Reply:
I think so. Somehow, weirdly it reads one of them as `\\n` and the other as `\n`

Reply:
So even if we split at `Counterexample`, `\\n` won't match `\n` when before `Counterexample`

Reply:
When I print `test.expectations` and `test.errors`:
```
[
  '// Warning 2529: (82-89): CHC: Empty array "pop" happens here.\\nCounterexample:\\na = []\\n\\n\\n\\nTransaction trace:\\nconstructor()\\nState: a = []\\nf()'
]
[
  'CHC: Empty array "pop" happens here.\n' +
    'Counterexample:\n' +
    'a = []\n' +
    '\n' +
    '\n' +
    '\n' +
    'Transaction trace:\n' +
    'constructor()\n' +
    'State: a = []\n' +
    'f()'
]
```